lib/proxy/ObjectCallProxy.js

```diff
@@ -0,0 +1,116 @@
+var _ = require('lodash');
+
+/**
+ * Create a new ObjectCallProxy.
+ *
+ * @class
+ */
+function ObjectCallProxy (source) {
+  this.source = source;
+  this.attached = {};
+  this.pending = [];
+}
+
+/**
+ * Restore all of the attached methods.
+ *
+ * @private
+ */
+ObjectCallProxy.prototype._restore = function () {
+  for (var name in this.attached) {
+    this.source[name] = this.attached[name];
+    delete this.attached[name];
+  }
+};
+
+/**
+ * Release all of the pending callbacks.
+ *
+ * @private
+ * @param {Error} [err] - Error to release all callbacks with (optional).
+ */
+ObjectCallProxy.prototype._release = function (err) {
+  var source = this.source;
+  var pending = this.pending;
+
+  this.pending = [];
+
+  pending.forEach(function (call) {
+    var args = call.args,
```

Comment:
Any reason for why you are using joined declarations here and not anywhere else? ;)